src/Compilers/CSharp/Portable/Symbols/NamedTypeSymbol.cs

```diff
@@ -374,6 +374,15 @@ internal void DoGetExtensionMethods(ArrayBuilder<MethodSymbol> methods, string n
                     if (method.IsExtensionMethod &&
                         ((options & LookupOptions.AllMethodsOnArityZero) != 0 || arity == method.Arity))
                     {
+                        var thisParam = method.Parameters.First();
```

Comment:
Yes. I cannot think of a way `method.IsExtensionMethod` can return true without a receiver parameter existing on the method.